opencaesar/oml-vision
# Feat/31-ui-fuseki-status

## Checklist before submitting a merge request


## Description of contribution

Add ability to see the ping and load status of a RDF triple store.

closes https://github.com/opencaesar/oml-vision/issues/31
 
## Testing performed




## How to Test Expected functionality changes

OML Model:
1. You can use this OML model for testing https://github.com/pogi7/kepler16b-example
2.  If not already done so, update the `src/vision/config/sparqlConfig.json` file to include a pingEndpoint.  Here's an example:
```json
{
  "queryEndpoint": "http://localhost:3030/tutorial2-inf/sparql",
  "updateEndpoint": "http://localhost:3030/tutorial2-inf/update",
  "pingEndpoint": "http://localhost:3030/$/ping"
}
```
3. Click the OML Vision eye icon
4. At first you will see a new window called `RDF Triplestore Status`.  Both icons should be red.
5. Once the gradle tasks load, click on the `Run startFuseki` task in the `Setup Tasks` window.
6. The `PING STATUS` should turn green
7. At this point click on the `Check if triplestore is loaded` button in the `RDF Triplestore Status` window.
8. The `LOAD STATUS` should still be red.  This is expected since we didn't load any data into the database yet.
9. From the terminal, run `./gradlew owlLoad`
10. When it finishes click on the `Check if triplestore is loaded` button in the `RDF Triplestore Status` window.
11. The `LOAD STATUS` will now turn green indicating that the triple store has data.
12. Click on the `Check if triplestore is loaded` button in the `RDF Triplestore Status` window.
13. You will see a popup in the lower right saying that `Triplestore has data!`

## Additional context

![image](https://github.com/opencaesar/oml-vision/assets/108915844/29a66def-91f5-4aa6-b2a9-0ad73c144a1f)
